module/VuFind/src/VuFind/RecordDriver/AbstractBase.php

```diff
@@ -351,22 +371,54 @@ public function getCitationFormats()
     {
         $formatSetting = $this->mainConfig->Record->citation_formats ?? true;
 
+        // Citations disabled:
+        if ($formatSetting === false || $formatSetting === 'false') {
+            return [];
+        }
+
         // Default behavior: use all supported options.
         if ($formatSetting === true || $formatSetting === 'true') {
-            return $this->getSupportedCitationFormats();
+            $formatSetting = $this->tryMethod(
+                'getDefaultCitationFormats',
+                [], // ← params, ↓ fallback to legacy
+                $this->getSupportedCitationFormats()
+            );
         }
 
-        // Citations disabled:
-        if ($formatSetting === false || $formatSetting === 'false') {
-            return [];
+        // Legacy: convert to array
+        if (is_string($formatSetting)) {
+            $formatSetting = explode(',', $formatSetting);
         }
 
-        // Filter based on include list:
-        $allowed = array_map('trim', explode(',', $formatSetting));
-        return array_intersect($allowed, $this->getSupportedCitationFormats());
+        // If no colon: convert to 9.x format
+        foreach ($formatSetting as $i => $format) {
+            if (strpos($format, ':') === false) {
+                $formatSetting[$i] = $this->expandLegacyAbbreviation($format);
+            }
+        }
+
+        // Trim and remove empty elements:
+        return array_filter(array_map('trim', $formatSetting));
```

Comment:
This seems to produce an error (tested with PHP 7.4):
`array_map(): Expected parameter 2 to be an array, object given`

Reply:
I think we need to do a `->toArray()` at the top of the method where the configuration is being loaded to avoid this.

Reply:
Took me a little time but I think I covered all the cases for this. Thank you!